### ISSUE TITLE
After rolling back node from 4.6 -> 4.5, scylla 4.5 was not started: Startup failed: std::_Nested_exception<no_such_column_family> (Can't find a column family with UUID: std::out_of_range (_Map_base::at)

### ISSUE THREAD

**Issue by aleksbykov** [scylladb/scylla#9092]
Kernel version: 4.18.0-305.10.2.el8_4.x86_64
Test-id: 42d85d9c-029c-4dbe-b846-83deb608d12a
ScyllaDB version: 4.5.rc5-0.20210711.7f96719c5 with build-id ecd5b049d7c2ba0db35d09c1dacfeccc6f804172 (-)
Target ScyllaDB repo: http://downloads.scylladb.com/unstable/scylla/master/rpm/centos/2021-07-23T01:10:18Z/scylla.repo
Target ScyllaVersion: 4.6.dev-0.20210722.a004b1da3
Instance type: n1-highmem-8
Number of ScyllaDB nodes: 4

Upgrade from scylla version 4.5 to 4.6 was successfully finished on node1. But next error appeared in log:
```
2021-07-23T06:03:34+00:00  rolling-upgrade-master-centos-db-node-42d85d9c-0-1 !ERR     | scylla[116080]: [shard 0] storage_proxy - Exception when communicating with 10.142.15.223, to read from system_auth.role_attributes: std::runtime_error (Failed to load schema version 91f8927a-10ff-39a9-9761-48f3500dfd29)
2021-07-23T06:03:34+00:00  rolling-upgrade-master-centos-db-node-42d85d9c-0-1 !ERR     | scylla[116080]: [shard 0] storage_proxy - Exception when communicating with 10.142.15.220, to read from system_auth.role_attributes: std::runtime_error (Failed to load schema version 91f8927a-10ff-39a9-9761-48f3500dfd29)

2021-07-23T06:03:34+00:00  rolling-upgrade-master-centos-db-node-42d85d9c-0-1 !ERR     | scylla[116080]: [shard 0] storage_proxy - Exception when communicating with 10.142.15.223, to read from system_auth.role_attributes: std::runtime_error (Failed to load schema version 91f8927a-10ff-39a9-9761-48f3500dfd29)
2021-07-23T06:03:34+00:00  rolling-upgrade-master-centos-db-node-42d85d9c-0-1 !ERR     | scylla[116080]: [shard 0] storage_proxy - Exception when communicating with 10.142.15.220, to read from system_auth.role_attributes: std::runtime_error (Failed to load schema version 91f8927a-10ff-39a9-9761-48f3500dfd29)
```
The errors above appeared in logs till the end of test.

Then node 3 was ugraded from 4.5 -> 4.6. Upgrade finished without errors. after Scylla 4.6 was started same error appeared in log2:
```
2021-07-23T06:31:49+00:00  rolling-upgrade-master-centos-db-node-42d85d9c-0-3 !ERR     | scylla[123060]: [shard 0] storage_proxy - Exception when communicating with 10.142.15.223, to read from system_distributed.service_levels: std::runtime_error (Failed to load schema version f9b316ca-0685-342d-906a-4b7dbaaffe08)
```

next step is roll back scylla from 4.6 -> 4.5. After rolling back, scylla 4.5 was not started due to next error:
```
2021-07-23T06:54:45+00:00  rolling-upgrade-master-centos-db-node-42d85d9c-0-3 !INFO    | scylla[135333]: [shard 0] init - Shutting down API server
2021-07-23T06:54:45+00:00  rolling-upgrade-master-centos-db-node-42d85d9c-0-3 !INFO    | scylla[135333]: [shard 0] init - Shutting down API server was successful
2021-07-23T06:54:45+00:00  rolling-upgrade-master-centos-db-node-42d85d9c-0-3 !INFO    | scylla[135333]: [shard 0] init - Shutting down tracing instance
2021-07-23T06:54:45+00:00  rolling-upgrade-master-centos-db-node-42d85d9c-0-3 !INFO    | scylla[135333]: [shard 0] init - Shutting down tracing instance was successful
2021-07-23T06:54:45+00:00  rolling-upgrade-master-centos-db-node-42d85d9c-0-3 !INFO    | scylla[135333]: [shard 0] init - Shutting down migration manager notifier
2021-07-23T06:54:45+00:00  rolling-upgrade-master-centos-db-node-42d85d9c-0-3 !INFO    | scylla[135333]: [shard 0] init - Shutting down migration manager notifier was successful
2021-07-23T06:54:45+00:00  rolling-upgrade-master-centos-db-node-42d85d9c-0-3 !INFO    | scylla[135333]: [shard 0] init - Shutting down prometheus API server
2021-07-23T06:54:45+00:00  rolling-upgrade-master-centos-db-node-42d85d9c-0-3 !INFO    | scylla[135333]: [shard 0] init - Shutting down prometheus API server was successful
2021-07-23T06:54:45+00:00  rolling-upgrade-master-centos-db-node-42d85d9c-0-3 !INFO    | scylla[135333]: [shard 0] init - Shutting down sighup
2021-07-23T06:54:45+00:00  rolling-upgrade-master-centos-db-node-42d85d9c-0-3 !INFO    | scylla[135333]: [shard 0] init - Shutting down sighup was successful
2021-07-23T06:54:45+00:00  rolling-upgrade-master-centos-db-node-42d85d9c-0-3 !ERR     | scylla[135333]: [shard 0] init - Startup failed: std::_Nested_exception<no_such_column_family> (Can't find a column family with UUID 5bc52802-de25-35ed-aeab-188eecebb090): std::out_of_range (_Map_base::at)
```

all db logs: 
https://cloudius-jenkins-test.s3.amazonaws.com/42d85d9c-029c-4dbe-b846-83deb608d12a/20210723_080917/db-cluster-42d85d9c.tar.gz

No coredumps

**Comment by bhalevy** [comment]
@tgrabiec can you please look into this issue?

**Comment by bhalevy** [comment]
@denesb does that ring a bell?

**Comment by denesb** [comment]
> @denesb does that ring a bell?

Not really but I will look into it.

**Comment by bhalevy** [comment]
@aleksbykov can we please try to reproduce with latest 4.5 and 4.6?

Cc @roydahan

**Comment by aleksbykov** [comment]
@bhalevy it is not yet reproduced. Once it happened i will update the issue

**Comment by bhalevy** [comment]
> @bhalevy it is not yet reproduced. Once it happened i will update the issue

@slivne I suggest we remove the `high` label then.

**Comment by slivne** [comment]
I agree removed it for now

**Comment by slivne** [comment]
Closing this issue - we released 4.6.0 / 4.6.1 -if the issue reproduces please post and reopen.
